File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        stack<string> st;
        string s2="";
        int noOfOpening=0;
        for(char c:s){
            if(c==')'){
                string temp;
                while(st.top()!="("){
                    string s1=st.top();
                    reverse(s1.begin(),s1.end());
temp+=s1;
st.pop();
                }

                st.pop();
                noOfOpening--;
                st.push(temp);
                if(noOfOpening==0){
                    s2+=temp;
                }
            }
            else{
                string s1;
        if(noOfOpening==0 && c!='('){
            s2+=c;
        }
        else if(c=='('){
                s1.push_back(c);
                st.push(s1);
                noOfOpening++;

        }
        else{
              s1.push_back(c);
                st.push(s1);
        }
            }
        }
      
        
        return s2;
    }
};

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.java
class Solution {
    public static String reverseParentheses(String s) {
    StringBuffer updatedString=new StringBuffer(s);
    int n=s.length();
    for (int i=0;i<n;i++) {
        if(s.charAt(i)==')'){
            int lastIndexOf=updatedString.substring(0, i).lastIndexOf("(");
           
            StringBuffer s1=new StringBuffer(updatedString.substring(lastIndexOf+1,i));
          
            s1.reverse();
        updatedString.replace(lastIndexOf+1, i, s1.toString());
        updatedString.setCharAt(lastIndexOf, 'A');
        updatedString.setCharAt(i, 'A');
       



        }
    }
    
    s=updatedString.toString();
   return s.replaceAll("A", "");
     
    }
}